Util/src/main/java/io/deephaven/util/reference/CleanupReferenceProcessor.java

```diff
@@ -98,29 +105,98 @@ public CleanupReferenceProcessor(
      *         {@link CleanupReferenceProcessor} instance
      */
     public <RT> ReferenceQueue<RT> getReferenceQueue() {
-        ReferenceQueue localQueue;
-        if ((localQueue = referenceQueue) == null) {
+        return getDrainQueue().referenceQueue();
+    }
+
+    private DrainQueue getDrainQueue() {
+        DrainQueue localQueue;
+        if ((localQueue = drainQueue) == null) {
             synchronized (this) {
-                if ((localQueue = referenceQueue) == null) {
-                    referenceQueue = localQueue = new ReferenceQueue<>();
-                    cleanerThread = new Thread(new DrainQueue(localQueue),
+                if ((localQueue = drainQueue) == null) {
+                    drainQueue = localQueue = new DrainQueue(new ReferenceQueue<>());
+                    cleanerThread = new Thread(localQueue,
                             "CleanupReferenceProcessor-" + name + "-drainingThread");
                     cleanerThread.setDaemon(true);
                     cleanerThread.start();
                 }
             }
         }
-        // noinspection unchecked
         return localQueue;
     }
 
+    /**
+     * Registers an object and a cleaning action to run when the object becomes phantom reachable.
+     *
+     * <p>
+     * The most efficient use is to explicitly invoke the cleanup method when the object is closed or no longer needed.
```

Comment:
Be explicit and provide a `{@link}` to the cleanup method you mean. We don't want the user mistakenly thinking that we've somehow made `action` exactly-once.